grpc/src/main/java/io/stargate/grpc/service/ValuesHelper.java

```diff
@@ -144,14 +166,34 @@ private static ResultSet processResult(Rows rows, boolean skipMetadata) throws S
       }
     }
 
+    int count = 0;
+
     for (List<ByteBuffer> row : rows.rows) {
+      ByteBuffer comparableBytes = null;
+      ByteBuffer rowPagingState = null;
+      if (makeRow != null) {
+        io.stargate.db.datastore.Row arrayListRow = makeRow.apply(columns, row);
+        comparableBytes = getComparableBytes.apply(columns, arrayListRow);
+        rowPagingState =
+            getPagingState.apply(
+                rows.resultMetadata.pagingState, arrayListRow, count == rows.rows.size() - 1);
+      }
```

Comment:
This would be an extra round of iteration of the `Rows` object, would that be an issue for larger ResultSets? For some reason I had thought that iterating once while the relevant List<ByteBuffer> representing the row was in scope would be most efficient, rather than scanning over it once to generate the enriched data and then scanning again to decode values and construct gRPC response.

Maybe it doesn't matter much though?